internal/indexer/indexer.go

```diff
@@ -111,6 +115,17 @@ func (i *Indexer) sync() (err error) {
 		if process_err != nil {
 			return process_err
 		}
+
+		counter++
+
+		// prometheus metics
+		metrics.IndexedBlocksCounter.Inc()
+		metrics.AverageBlockProcessingTimeMilisec.Set(float64((time.Since(prev).Milliseconds() / counter)))
+		metrics.IndexedBlocksTotal.Set(float64(chain.ParseInt(blk.Number)))
+		metrics.IndexedDagBlocksTotal.Set(float64(finalizationData.DagCount))
+		metrics.IndexedPbftBlocksTotal.Set(float64(finalizationData.PbftCount))
```

Comment:
Looks like it is should be the same thing as `IndexedBlocksTotal`
This code will set just the same data here every time as you getting `finalizationData` from db before this for cycle. Should be:
```
		metrics.IndexedPbftBlocksTotal.Set(float64(p))
		metrics.IndexedDagBlocksTotal.Set(float64(dc))
		metrics.IndexedTransactionsTotal.Set(float64(tc))
```
Also this `AverageBlockProcessingTimeMilisec` isn't set correctly. As `prev` is reset every 100 blocks. So you should set counter near [this line](https://github.com/Taraxa-project/taraxa-indexer/pull/54/files#diff-87c2475baa87005bd3a3001ec33f201feea23caddbbf5a2f249238a2bd32026dL107) also to get correct data here

Reply:
And also we are processing blocks not only during syncing. So you need to set all this data here https://github.com/Taraxa-project/taraxa-indexer/pull/54/files#diff-87c2475baa87005bd3a3001ec33f201feea23caddbbf5a2f249238a2bd32026dR173
Or maybe even better will be just to move it inside of this `process` method